crates/ty_python_semantic/src/types/infer/builder.rs

```diff
@@ -4895,7 +4895,9 @@ impl<'db, 'ast> TypeInferenceBuilder<'db, 'ast> {
             self.infer_type_expression(&bound.value);
         }
         if let Some(default) = arguments.find_keyword("default") {
-            let func_ty = self.get_or_infer_expression(func, TypeContext::default());
+            let func_ty = self
+                .try_expression_type(func)
+                .unwrap_or_else(|| self.infer_expression(func, TypeContext::default()));
             if func_ty.as_class_literal().is_some_and(|class_literal| {
                 class_literal.is_known(self.db(), KnownClass::ParamSpec)
             }) {
```

Comment:
I think at this point, the expression should've already been inferred because this is in the deferred region and the function symbol should've already been inferred. So, I could directly use `self.expression_type` instead.